### PR TITLE
Improvments for the README and Download page

### PR DESCRIPTION
**Pull request by chibenwa**
## Why

GitHub is an important place to get traction and READMEs are the primary document people look at.

They should be both informative AND easy to read.

My personnal feeling is that James README is polluted by too much information, not readable and misses most of the points.

The download page suffers similar flows.

I would like to improve the current situation with this humble PR.

## How

The goal is to keep the same level of information on the README but link external pages rather than polluting the README with content not pertinant for most readers.

Also I propose using badges at the top for important links: download, license, website, CI, documentation, docker hub, websites posts. Hopefully these contents would get more visibility.

I drastically simplified the index in the README that was clearly overcharged to me (takes more than my screen!) thus unreadable.

Finally I also sanitised the download page as stated in https://www.mail-archive.com/server-dev@james.apache.org/msg71016.html

## Preview

Preview these changes on my personnal repository: https://github.com/chibenwa/james-project/tree/improve-reame-14-09-2021

### CODE REVIEW

**Comment by chibenwa** [comment]
https://github.com/chibenwa/james-project/tree/improve-reame-14-09-2021-2 uses collapsible sections to get a really cons=cise easy to read readme and get rid of the index section.

(cc @rouazana  maybe? )

File: docs/modules/development/pages/deployment-tests.adoc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+= How to run deployment Tests
+:navtitle: Deployment tests
+
+We wrote some MPT (James' Mail Protocols Tests subproject) deployment tests to validate a James
+deployment.
+
+It uses the External-James module, that uses environment variables to locate a remote
+IMAP server and run integration tests against it.
+
+For that, the target James Server needs to be configured with a domain domain and a user imapuser
```
Comment (vttranlina) [inline docs/modules/development/pages/deployment-tests.adoc:10]: double `domain`

File: docs/modules/development/pages/deployment-tests.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+with password password. Read above documentation to see how you can do this.
```
Comment (vttranlina) [inline docs/modules/development/pages/deployment-tests.adoc:11]: double `password`

File: docs/modules/development/pages/deployment-tests.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+You have to run MPT tests inside docker. As you need to use maven, the simplest option is to
+use james/parent image, and override the entry point ( as git and maven are already configured
+there ) :
+
+    $ export JAMES_ADDRESS=127.0.0.1
+    $ export JAMES_IMAP_PORT=143
+    $ export JAMES_SMTP_PORT=25
+
+    $ mvn -T 1C -DskipTests -pl org.apache.james:apache-james-mpt-external-james -am install
+    $ mvn -T 1C -pl org.apache.james:apache-james-mpt-external-james test -Pintegration-tests\
```
Comment (vttranlina) [inline docs/modules/development/pages/deployment-tests.adoc:22]: remove " \ "

File: README.adoc [hunk 4]
```diff
@@ -93,7 +93,7 @@ Useful options includes:
 
 == How to run James in Docker
 
-This feature is available for three configurations :
+We maintain docker distribution for our Guice based applications :
```
Comment (vttranlina) [inline README.adoc:97]:
```suggestion
We maintain docker distribution for our link:https://github.com/google/guice[Guice] based applications:
```

File: README.adoc [hunk 5]
```diff
@@ -5,7 +5,7 @@ image:https://img.shields.io/badge/Join%20us-Mailing%20lists-purple.svg[link="ht
 link:https://gitter.im/apache/james-project[image:https://badges.gitter.im/apache/james-project.svg[Join the chat at link:https://gitter.im/apache/james-project]]
 image:https://img.shields.io/badge/CI-Jenkins%40Apache-blue.svg[link="https://ci-builds.apache.org/job/james/job/ApacheJames/"]
 image:https://img.shields.io/badge/Documentation-green.svg[link="https://james.apache.org/documentation.html"]
-image:https://img.shields.io/badge/Downloads-yellow.svg[link="https://james.apache.org/download.cgi"]
+image:https://img.shields.io/badge/Downloads-3.6.0-yellow.svg[link="https://james.apache.org/download.cgi"]
```
Comment (vttranlina) [inline README.adoc:8]: Should we use https://archive.apache.org/dist/james/server/3.6.0/ ?

**Review by vttranlina — COMMENTED** [review]
Good job

File: README.adoc (hunk 5 above)
Comment (chibenwa) [reply README.adoc:8]: Don't point directly to Apache hosting, we should encourage using the mirrors

File: docs/modules/development/pages/deployment-tests.adoc (hunk 1 above)
Comment (chibenwa) [inline docs/modules/development/pages/deployment-tests.adoc:10]:
```suggestion
For that, the target James Server needs to be configured with a domain `domain` and a user `imapuser`
```

File: docs/modules/development/pages/deployment-tests.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+:navtitle: Deployment tests
+
+We wrote some MPT (James' Mail Protocols Tests subproject) deployment tests to validate a James
+deployment.
+
+It uses the External-James module, that uses environment variables to locate a remote
+IMAP server and run integration tests against it.
+
+For that, the target James Server needs to be configured with a domain `domain` and a user `imapuser`
+with password password. Read above documentation to see how you can do this.
```
Comment (chibenwa) [inline docs/modules/development/pages/deployment-tests.adoc:11]:
```suggestion
with password `password`. Read above documentation to see how you can do this.
```

File: docs/modules/development/pages/deployment-tests.adoc [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+You have to run MPT tests inside docker. As you need to use maven, the simplest option is to
+use james/parent image, and override the entry point ( as git and maven are already configured
+there ) :
+
+    $ export JAMES_ADDRESS=127.0.0.1
+    $ export JAMES_IMAP_PORT=143
+    $ export JAMES_SMTP_PORT=25
+
+    $ mvn -T 1C -DskipTests -pl org.apache.james:apache-james-mpt-external-james -am install
+    $ mvn -T 1C -pl org.apache.james:apache-james-mpt-external-james test -Pintegration-tests\
```
Comment (chibenwa) [inline docs/modules/development/pages/deployment-tests.adoc:22]:
```suggestion
    $ mvn -T 1C -pl org.apache.james:apache-james-mpt-external-james test -Pintegration-tests
```

File: README.adoc [hunk 8]
```diff
@@ -15,15 +20,21 @@ It has a modular architecture based on a rich set of *modern* and *efficient* co
 Create your *own personal solution* for emails treatment by assembling the components you need thanks to the Inversion
 of Control mail platform offered and  go further customizing filtering and routing rules using *James Mailet Container*.
 
-Supported protocols are: IMAP, SMTP, JMAP, POP3 and more...
+Supported protocols are: IMAP, SMTP, link:https://jmap.io[JMAP], POP3 and more...
+
+The link:https://james.staged.apache.org/james-project/3.7.0/servers/distributed.html[*Distributed server*] focuses on delivering a easy-to-operate scalable mail server based on modern technologies
```
Comment (Arsnael) [inline README.adoc:25]:
```suggestion
The link:https://james.staged.apache.org/james-project/3.7.0/servers/distributed.html[*Distributed server*] focuses on delivering an easy-to-operate scalable mail server based on modern technologies
```

File: docs/modules/servers/partials/ManageSieveMailet.adoc [hunk 9]
```diff
@@ -0,0 +1,13 @@
+=== Manage Sieve scripts via a mailet
+
+Each user can manage his SIEVE scripts through the `ManageSieveMailet` mailet.
+
+To use the manage SIEVE mailet :
+
+* You need to create the user sievemanager@DOMAIN ( if you don't, the SMTP server will check the domain, recognize it, and look for an absent local user, and will generate an error ).
+* You can send Manage Sieve commands by mail to sievemanager@DOMAIN. Your subject must contain the command. Scripts needs to be added as attachments and need the ".sieve" extension.
```
Comment (Arsnael) [inline docs/modules/servers/partials/ManageSieveMailet.adoc:8]:
```suggestion
* You can send Manage Sieve commands by mail to sievemanager@DOMAIN. Your subject must contain the command. Scripts need to be added as attachments and need the ".sieve" extension.
```

**Review by Arsnael — COMMENTED** [review]
Good initiative, thanks :) Just a few minor comments

**Review by Arsnael — APPROVED** [review]

File: src/site/markdown/server/manage-cli.md [hunk 10]
```diff
@@ -15,6 +15,12 @@ And for Guice distributions:
 java -jar /root/james-cli.jar -h 127.0.0.1 -p 9999 COMMAND
 ```
 
+Guice in docker embed a script helper:
```
Comment (rouazana) [inline src/site/markdown/server/manage-cli.md:18]: embeds? (suggestions are broken?)

File: README.adoc [hunk 11]
```diff
@@ -57,30 +57,24 @@ There is many other ways one can help us: packaging, communication, etc ...
 
 == How to try James
 
-Requirements: docker & docker-compose installed.
+Requirements: `docker` installed.
 
-When you try James this way, you will use the most current state of James.
-It will be configured to run with Cassandra & ElasticSearch.
-All those three components will be started with a single command.
+Here you will try James server v 3.6.0 thanks to a docker image. This James image has a default configuration using JPA
```
Comment (rouazana) [inline README.adoc:59]: v3.6.0 or version 3.6.0?

File: README.adoc [hunk 12]
```diff
@@ -57,30 +57,24 @@ There is many other ways one can help us: packaging, communication, etc ...
...
+(hsqldb) and Lucene. It also includes a default domain named james.local and three default users: user01@james.local,
+user02@james.local, user03@james.local, with their default password being 1234.
 
-You can retrieve the docker-compose file :
+Note: this James server will respond to IMAPS port 993 and SMTPS port 465.
 
-    $ wget https://raw.githubusercontent.com/apache/james-project/master/server/apps/distributed-app/docker-compose.yml
+Pull and run the James image with the following single command :
```
Comment (rouazana) [inline README.adoc:65]: s/command :/command:/

File: README.adoc [hunk 13]
```diff
@@ -57,30 +57,24 @@ There is many other ways one can help us: packaging, communication, etc ...
...
 
-Then, you just have to start the services:
+    $ docker run -p "465:465" -p "993:993" apache/james:demo-3.6.0
 
-    $ docker-compose up
+Then, connect this image with for instance, Thunderbird.
+link:https://james.apache.org/howTo/imap-server.html[This tutorial] covers more in depth user and domain creation, as well as thunderbird setup.
```
Comment (rouazana) [inline README.adoc:70]: s/thunderbird/Thunderbird/

**Review by rouazana — APPROVED** [review]
nice rework!

File: README.adoc [hunk 14]
```diff
@@ -36,173 +47,81 @@ https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolut
...
-All those three components will be started with a single command.
-
-You can retrieve the docker-compose file :
+.Read more...
+[%collapsible]
+====
+Requirements: `docker` installed.
 
-    $ wget https://raw.githubusercontent.com/apache/james-project/master/server/apps/distributed-app/docker-compose.yml
+Here you will try James server v 3.6.0 thanks to a docker image. This James image has a default configuration using JPA
```
Comment (chibenwa) [inline README.adoc:59]:
```suggestion
Here you will try James server version 3.6.0 thanks to a docker image. This James image has a default configuration using JPA
```

File: README.adoc [hunk 15]
```diff
@@ -36,173 +47,81 @@ https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolut
...
+(hsqldb) and Lucene. It also includes a default domain named james.local and three default users: user01@james.local,
+user02@james.local, user03@james.local, with their default password being 1234.
 
-Then, you just have to start the services:
+Note: this James server will respond to IMAPS port 993 and SMTPS port 465.
 
-    $ docker-compose up
+Pull and run the James image with the following single command :
```
Comment (chibenwa) [inline README.adoc:65]:
```suggestion
Pull and run the James image with the following single command:
```

File: README.adoc [hunk 16]
```diff
@@ -36,173 +47,81 @@ https://issues.apache.org/jira/issues/?jql=project%20%3D%20JAMES%20AND%20resolut
...
 
-Wait a few seconds in order to have all those services start up. You will see the following log when James is available:
-james           | Started : true
+    $ docker run -p "465:465" -p "993:993" apache/james:demo-3.6.0
 
-A default domain, james.local, has been created. You can see this by running:
+Then, connect this image with for instance, Thunderbird.
+link:https://james.apache.org/howTo/imap-server.html[This tutorial] covers more in depth user and domain creation, as well as thunderbird setup.
```
Comment (chibenwa) [inline README.adoc:70]:
```suggestion
link:https://james.apache.org/howTo/imap-server.html[This tutorial] covers more in depth user and domain creation, as well as Thunderbird setup.
```

File: src/site/markdown/server/manage-cli.md (hunk 10 above)
Comment (chibenwa) [inline src/site/markdown/server/manage-cli.md:18]:
```suggestion
Guice in docker embeds a script helper:
```

**Review by Arsnael — APPROVED** [review]

**Pull request merged**